bin/varnishd/cache/cache_wrk.c

```diff
@@ -160,28 +160,39 @@ WRK_Thread(struct pool *qp, size_t stacksize, unsigned thread_workspace)
  * Summing of stats into pool counters
  */
 
-static void
-pool_addstat(struct VSC_main_wrk *dst, struct VSC_main_wrk *src)
+static unsigned
+wrk_addstat(struct worker *wrk, const struct pool_task *tp, unsigned locked)
 {
+	struct pool *pp;
 
-	dst->summs++;
-	VSC_main_Summ_wrk_wrk(dst, src);
-	memset(src, 0, sizeof *src);
+	CHECK_OBJ_NOTNULL(wrk, WORKER_MAGIC);
+	pp = wrk->pool;
+	CHECK_OBJ_NOTNULL(pp, POOL_MAGIC);
+	if (locked)
+		Lck_AssertHeld(&pp->mtx);
+
+	if ((tp == NULL && wrk->stats->summs > 0) ||
```

Comment:
I think we should move this test `CHECK_OBJ_NOTNULL(wrk, WORKER_MAGIC);` and return unless true to avoid useless assertion tests on `pp`

Reply:
I didn't want to do anything before making all assertions, the current order was chosen on purpose.